### ISSUE TITLE
[BUG] Download failed

### ISSUE THREAD

**Issue by domjeandom** [Tzahi12345/YoutubeDL-Material#847]
Hi everyone
It's the first time I report a bug and sorry if I don't respect rules. 
Thank for your help to solve this issue.

Thanks a lot 
Dominique

**Describe the bug**
Impossible to download a video from youtube. 

**To Reproduce**

Any time

**Expected behavior**

I use YoutubeDL for few years without problem, I am not sure but it's possible this issue is arrived after dsm update 7.1 update 4

**Screenshots**
Error while retrieving info on video with URL https://www.youtube.com/watch?v=aLdGo1XCwsM with the following message: Error: Command failed with exit code 1: node_modules/youtube-dl/bin/youtube-dl -o users/dominique/video/%(title)s.mp4 --write-info-json --print-json -f bestvideo+bestaudio --merge-output-format mp4 --write-thumbnail --dump-json http://www.youtube.com/watch?v=aLdGo1XCwsM

ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

**Environment**
- YoutubeDL-Material       version 4.3
- Docker tag: <tag> (optional)

Ideally you'd copy the info as presented on the "About" dialogue
in YoutubeDL-Material.
Installation details: 

Installed version: v4.3 -   You are up to date.
Installation type: docker
Docker tag: latest
Commit hash: fc3c179
Build date: 2022-07-05

**Additional context**
Logs:
<html>
<body>
<div class="x-grid3-row x-grid3-row-first x-grid3-row-selected" id="ext-gen5302" style="margin: 0px; padding: 0px; cursor: default; border-width: 0px 0px 1px; border-style: dotted dotted solid; border-top-color: initial; border-right-color: initial; border-bottom-color: rgba(198, 212, 224, 0.4); border-left-color: initial; border-image: initial; width: 1949px; background-color: rgba(5, 127, 235, 0.1) !important; color: rgb(65, 75, 85); font-size: 13px; font-family: Verdana, Arial, sans-serif; font-style: normal; font-variant-ligatures: none; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; height: 24px;">

2023-03-16T10:28:25: PM2 log: [PM2] This PM2 is not UP TO DATE | stdout
-- | --

</div>
</body>
</html>

**Comment by rdragomir** [comment]
Same issue here. I'm using DSM 6.2.4-25556 Update 6 and latest Youtube Downloader.
I tried to DL the audio track of 6 different youtube videos and with all of them I had the same issue and they all failed.

**Comment by imjustabill** [comment]
I ran into this issue, and it was due to youtube-dl being out of date. Make sure your youtube-dl is updated, or update to the latest container if you're running via docker.

**Comment by Sharmn22** [comment]
I ran into this issue this morning. 
v4.3
Docker installed on Synology NAS  DSM 7.1.1-42962 Update 3 


```
ERROR: [youtube] pY0LnKiDwRA: Unable to extract uploader id; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
2023-03-23T07:06:45.994Z ERROR: Subscribe: Failed to get subscription info. Subscribe failed. | stdout
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined

 at new NodeError (node:internal/errors:371:5)
    at validateString (node:internal/validators:120:11)
    at Object.join (node:path:1172:7)
    at /app/app.js:1533:34
    at Layer.handle [as handle_request] (/app/node_modules/express/lib/router/layer.js:95:5)
    at next (/app/node_modules/express/lib/router/route.js:137:13)
    at optionalJwt (/app/app.js:739:12)
    at Layer.handle [as handle_request] (/app/node_modules/express/lib/router/layer.js:95:5)
    at next (/app/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/app/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/app/node_modules/express/lib/router/layer.js:95:5)
    at /app/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/app/node_modules/express/lib/router/index.js:341:12)
    at next (/app/node_modules/express/lib/router/index.js:275:10)
    at compression (/app/node_modules/compression/index.js:220:5)
    at Layer.handle [as handle_request] (/app/node_modules/express/lib/router/layer.js:95:5) {
  code: 'ERR_INVALID_ARG_TYPE'
}

**Comment by martadinata666** [comment]
go to task and update the youtube-dl binary from there

**Comment by domjeandom** [comment]
Hi Martadinata666,

Thank for your reply
Can you be more explicit.. like how to do.
In my mind if I am at the latest image tzahi12345/youtubedl-material I am upto date for every needed to work well.
I must update youtube-dl binary in the container via terminal in the container??? right? the same question How to do!!
I am a simple user with low skill in LINUX.

Thank in advance for your reply.
Dominique

**Comment by martadinata666** [comment]
![Screenshot from 2023-03-23 17-57-12](https://user-images.githubusercontent.com/2433562/227182735-4a8f433d-836b-41b5-9fdb-5c68b13983fd.png)
I don't know how syno manage upgrade or the image, that is my youtubedl-material web page, the image is `tzahi12345/youtubedl-material:latest`. I deployed over docker-compose.

and if you can go to container terminal. the youtube-dl binary at `/app/node_modules/youtube-dl/bin`

**Comment by Sharmn22** [comment]
It works well now - I have updated NPM from Synology CLI using SSH `sudo npm install -g npm@9.6.2`

Still It doesn't download a new Playlist if I add it there under Subscriptions section. I can live with that :)

**Comment by aerodb** [comment]
I have forced the docker container to update. as well as visited the tasks tab and selected to run all the tasks. however, once completed, there was no change to result when i attempt to download a video. I did notice that under the tasks tab, only the missing files check shows a last confirmed date when i ran all the tasks at least twice. does this indicate that the update task didnt complete? I did get a message that it completed successfully when i selected it. an updated error message will be provided below. 

Error while retrieving info on video with URL https://www.youtube.com/watch?v=REDACTED with the following message: Error: Command failed with exit code 1: node_modules/youtube-dl/bin/youtube-dl -o subscriptions/playlists/REDACTED/%(upload_date)s - %(title)s.%(ext)s --write-info-json --print-json -f bestvideo+bestaudio --merge-output-format mp4 --write-thumbnail --write-auto-sub --dump-json http://www.youtube.com/watch?v=REDACTED ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type youtube-dl -U to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

**Comment by martadinata666** [comment]
![Screenshot from 2023-03-23 21-37-50](https://user-images.githubusercontent.com/2433562/227237814-7aecc1e9-72db-4908-9ac3-2b40d065cd06.png)

I tried from scratch, I check "press play" button and it show update binary to `2023.03.04`.

**Comment by domjeandom** [comment]
same for me like aerodb. Update complete but ngm version stay the same
![npm](https://user-images.githubusercontent.com/128362951/227263194-8d9fdaca-d078-449d-9fa2-822369a25f65.jpg)

**Comment by CzechThisOut01** [comment]
Same exact issue. YouTube-DL up to date version 4.3, yet, unable to download any content on YouTube with result of same error as domjeandon original post. 


ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type youtube-dl -U to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
2023-03-24T01:09:27.444Z ERROR: Error while retrieving info on video with URL https://www.youtube.com/watch?v=S48Tc_LLHis&t=18s with the following message: Error: Command failed with exit code 1: node_modules/youtube-dl/bin/youtube-dl -o video/%(title)s.mp4 --write-info-json --print-json -f bestvideo+bestaudio --merge-output-format mp4 --dump-json http://www.youtube.com/watch?v=S48Tc_LLHis

**Comment by CzechThisOut01** [comment]
> same for me like aerodb. Update complete but ngm version stay the same ![npm](https://user-images.githubusercontent.com/128362951/227263194-8d9fdaca-d078-449d-9fa2-822369a25f65.jpg)

Hi!
 Any luck to make this work or we just screwed  :-/  
 The latest version is year old  :-o

**Comment by anguslilei** [comment]
I also encountered this problem, but I solved it. Here is the solution:

1. Click on the top left corner --> Settings --> Advanced, and choose the downloader "yt-dlp".Click SAVE. This step is very important as the default value may be "youtube-dl".
2. Go to Task --> Update yt-dlp, and click the Run button.
3. Normally, the problem should be solved by now. If not, please restart the service.

**Comment by CzechThisOut01** [comment]
It WORKED....I know I know you probably said about now....DUH...I know it does LOL....No seriously, THANK YOU!!!
 Again, knowledge prevailed..... 👍

**Comment by aerodb** [comment]
same, i suppose i was using a no longer supported downloader option. thank you @anguslilei

**Comment by Tzahi12345** [comment]
It sounds like the main issue is legacy instances (or configs) defaulting to `youtube-dl` which doesn't seem to work very well. I added a warning to suggest users switch to yt-dlp: https://github.com/Tzahi12345/YoutubeDL-Material/commit/3544a2316d74064011c39ec35267af89c4086568

This will be included in v4.3.1. If this is not why your downloads are failing let me know and I'll re-open this issue.
